Automattic/wp-calypso
# Adds a new origin for the promote post widget

## Proposed Changes

* Adds the calculation of the new origin for the Promote Post Widget.

The origin new name will be: `wp-blaze-plugin`

## Why are these changes being made?

We are working on making our Blaze Ads plugin compatible with non-woocommerce sites. As part of that change, we want to differentiate the origin of the campaigns when opening the widget so we can properly use feature flags and track on the DSP side.

## Testing Instructions

Until we release the new version of the Blaze Ads plugin, there won't be a direct way to test this. We can test using this alternative method that simulates how the plugin uses the Blaze Dashboard app.

* Checkout this branch
* In the `config/development.json` file, add a new feature called `is_running_in_blaze_plugin` with the value `true`
```
        ...,
	"features": {
		"is_running_in_blaze_plugin": true,
                ...
         }
```
* Start the server and Navigate to Tools->Advertising
* Open the browser console (we need to check in a request if we sent the correct origin to the widget)
* On the Advertising page, click the Promote button for any of your posts
* In the network console, verify the payload of the request `dsp/api/v1/user/check`. You should see the new origin `wp-blaze-plugin`
![Screenshot 2024-07-29 at 4 56 17 PM](https://github.com/user-attachments/assets/7fcdac16-e76a-4fc8-a0d4-d8a497e7a8d0)

## Pre-merge Checklist
